src/irmin-pack/unix/dispatcher.ml

```diff
@@ -240,7 +240,7 @@ module Make (Fm : File_manager.S with module Io = Io.Unix) :
       let accessor =
         Accessor.v_exn t ~off:entry_offset_suffix_start ~len:read_in_suffix
       in
-      read_exn t accessor buf;
+      read_exn t accessor buf_suffix;
       Bytes.blit buf_suffix 0 buf read_in_prefix read_in_suffix)
```

Comment:
Nice catch! (what's the story behind reading a range overlapping the prefix and suffix?)

Reply:
I will try an explanation (with this PR, not on main)

    prefix.1.                  suffix.1.
|---------------||--------------|
0 ---- 659 -- 700 -- 720 ---- 859 

The last object in prefix.1. is commit1. The first object in suffix.1 is blob2 referenced by commit2. We call gc on commit2, which has commit1 as the parent. 

We will call register_entry for both commit1 and blob2. When the intervals of offsets are merged for creating the mapping file, we end up with an interval [659, 720]. This implies both a read in the prefix and one in the suffix, in order to transfer that chunk of data to prefix.2.

Hope this helps, it's not super evident.